disperser/batcher/batcher.go

```diff
@@ -262,15 +262,8 @@ func (b *Batcher) updateConfirmationInfo(
 				blobsToRetry = append(blobsToRetry, batchData.blobs[blobIndex])
 				continue
 			}
-			blobHeader := batchData.blobHeaders[blobIndex]
 
-			blobHeaderHash, err := blobHeader.GetBlobHeaderHash()
-			if err != nil {
-				b.logger.Error("HandleSingleBatch: failed to get blob header hash", "err", err)
-				blobsToRetry = append(blobsToRetry, batchData.blobs[blobIndex])
-				continue
-			}
-			merkleProof, err := batchData.merkleTree.GenerateProof(blobHeaderHash[:], 0)
+			merkleProof, err := batchData.merkleTree.GenerateProofWithIndex(uint64(blobIndex), 0)
```

Comment:
Any compatibility issue with this change?

Reply:
Shouldn't be. It won't affect blobs that had been dispersed already. It will only affect blobs to be dispersed after the deploy by populating correct inclusion proofs in the confirmation info

Reply:
What about the consumer of this inclusion proof? Will this break anything on that side?

Reply:
No the consumer of the inclusion proof reads it from DB without re-generating it